src/components/social.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import Social from './social.js';
+import reactGA from 'react-ga';
+
+var ShareProgressButton = React.createClass({
+  componentDidMount: function() {
+    var shareProgressHolder = document.querySelector(".share-progress-holder");
+    var shareProgressButton = shareProgressHolder.querySelector("." + this.props.shareProgress);
+    if (shareProgressButton) {
+      shareProgressHolder.removeChild(shareProgressButton);
+      this.shareProgressContainer.appendChild(shareProgressButton);
+    }
+  },
+  componentWillUnmount: function() {
+    var shareProgressButton = this.shareProgressContainer.querySelector("." + this.props.shareProgress);
```

Comment:
as an unmount, we should be able to cache the `shareProgressButton` and `shareProgressHolder` in componentWillMount, rather than extracting them out of the DOM again?

Reply:
Everything works fine cept one change.

I cannot cleanly do this in componentWillMount, but instead componentDidMount

The reason is componentWillMount is run on both client and server. On the server, document won't exist.

componentDidMount however, is only a client side run function, so it's safe to do `document.` stuff in here.